### PR TITLE
UI/15496317 all members page

### PR DESCRIPTION
**Pull request by CzarekDryl**
Added card listing to members page

https://tw.auroracreation.com/app/tasks/15496317

image

### CODE REVIEW

File: src/components/v5/shared/CardWithBios/types.ts [hunk 1]
```diff
@@ -16,14 +16,13 @@ export type PermissionType =
...
-  extensionData: AnyExtensionData;
-  percentage?: number;
   shouldStatusBeVisible: boolean;
   userStatus?: UserStatusMode;
   permissions?: Permissions[];
-  userStatusTooltipDetails: UserStatusTooltipDetailsProps;
+  userStatusTooltipDetails?: UserStatusTooltipDetailsProps;
   shouldBeMenuVisible: boolean;
   isVerified?: boolean;
+  userData?: Member | undefined;
```
Comment (krawczykkatarzyna) [inline src/components/v5/shared/CardWithBios/types.ts:25]: userData?: Member

File: src/components/v5/common/MembersList/MembersList.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  const { handleClipboardCopy } = useMembersList();
+
+  const listLength = list.length;
+
+  return (
+    <div>
+      <div className="flex items-center mb-2">
+        <h3 className="heading-5 mr-3">{formatMessage(title)}</h3>
+        <span className="text-md text-blue-400">
+          {listLength} {formatMessage(title)}
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/MembersList/MembersList.tsx:31]: should `listLength` always be visible?

File: src/components/v5/common/MembersList/MembersList.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        </span>
+      </div>
+      <p className="mb-6">{formatMessage(description)}</p>
+      {!isLoading && (
+        <ul>
+          {listLength > 0 ? (
+            list.map((item) => (
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/MembersList/MembersList.tsx:38]: please use destructuring assignment

File: src/components/v5/common/MembersList/MembersList.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+              title={emptyTitle}
+              description={emptyDescription}
+              withBorder
+              buttonText={{ id: 'members.subnav.invite' }}
+              onClick={handleClipboardCopy}
+            />
+          )}
+        </ul>
+      )}
+      {listLength > 0 && (
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/MembersList/MembersList.tsx:60]: `!!listLength`

File: src/components/v5/common/MembersList/MembersList.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      <div className="flex items-center mb-2">
+        <h3 className="heading-5 mr-3">{formatMessage(title)}</h3>
+        <span className="text-md text-blue-400">
+          {listLength} {formatMessage(title)}
+        </span>
+      </div>
+      <p className="mb-6">{formatMessage(description)}</p>
+      {!isLoading && (
+        <ul>
+          {listLength > 0 ? (
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/MembersList/MembersList.tsx:37]: `!!listLength`

File: src/components/v5/common/MembersList/MembersList.tsx (hunk 2 above)
Comment (CzarekDryl) [reply src/components/v5/common/MembersList/MembersList.tsx:31]: Yup, even when there are no members this shows 0

File: src/components/v5/common/MembersList/MembersList.tsx (hunk 3 above)
Comment (CzarekDryl) [reply src/components/v5/common/MembersList/MembersList.tsx]: Fixed

File: src/components/v5/common/MembersList/MembersList.tsx (hunk 4 above)
Comment (CzarekDryl) [reply src/components/v5/common/MembersList/MembersList.tsx]: Fixed

File: src/components/v5/common/MembersList/MembersList.tsx (hunk 5 above)
Comment (CzarekDryl) [reply src/components/v5/common/MembersList/MembersList.tsx]: It was meant to be listLength > 8 and not 0, so I've fixed that

File: src/components/frame/v5/pages/MembersPage/hooks.ts [hunk 6]
```diff
@@ -1,6 +1,13 @@
 import { usePopperTooltip } from 'react-popper-tooltip';
+import { useGetMembersForColonyQuery } from '~gql';
+import { useColonyContext } from '~hooks';
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/frame/v5/pages/MembersPage/hooks.ts:3]: Add a blank space, I think a good way is to keep separated by the blank line imports packages and other imports like components, hooks, types, styles, etc.

File: src/components/v5/common/MembersList/MembersList.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
+import React, { FC } from 'react';
+import { useIntl } from 'react-intl';
+
+import { MembersListProps } from './types';
+import CardWithBios from '~v5/shared/CardWithBios/CardWithBios';
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/v5/common/MembersList/MembersList.tsx:5]: import CardWithBios from '~v5/shared/CardWithBios';

File: src/components/v5/common/MembersList/MembersList.tsx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        <h3 className="heading-5 mr-3">{formatMessage(title)}</h3>
+        <span className="text-md text-blue-400">
+          {listLength} {formatMessage(title)}
+        </span>
+      </div>
+      <p className="mb-6">{formatMessage(description)}</p>
+      {!isLoading && (
+        <ul>
+          {listLength ? (
+            list.map((item) => {
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/v5/common/MembersList/MembersList.tsx:38]: `  list.map(({user}) => {`

Comment (CzarekDryl) [reply src/components/v5/common/MembersList/MembersList.tsx:38]: I have to pass whole `item` prop to `<CardWithBios />`

**Review by KatarzynaStefanowiczPagepro — APPROVED** [review]

**Pull request merged**